### DISCUSSION TITLE
Need some assistance

### DISCUSSION THREAD

**Discussion by FECRwy** [DCC-EX/EX-T3-WiFi#119] [General]
I'm building my second throttle and have run into some difficulty. I have tried loading the code both by the Platform.ino file and the new web-based install code. In both cases I've just got a flashing screen with the following repeated in the terminal window:

ELF file SHA256: d29581baf8070047

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4
[   214][E][esp32-hal-cpu.c:110] addApbChangeCallback(): duplicate func=0x40104738 arg=0x3ffbde68
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400e9ed3  PS      : 0x00060830  A0      : 0x80105201  A1      : 0x3ffb27d0  
A2      : 0x3ffc4800  A3      : 0x3ffc585c  A4      : 0x3ffc4e00  A5      : 0x00000055
A6      : 0x3ffb8874  A7      : 0x80000001  A8      : 0x800e9ed1  A9      : 0x3ffb27b0  
A10     : 0x00000000  A11     : 0x00000000  A12     : 0x00000033  A13     : 0x0000005d
A14     : 0x00061a80  A15     : 0x3ffc5024  SAR     : 0x0000001c  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000015  LBEG    : 0x40089eb4  LEND    : 0x40089eca  LCOUNT  : 0xffffffff

Backtrace: 0x400e9ed0:0x3ffb27d0 0x401051fe:0x3ffb2820

  #0  0x400e9ed0:0x3ffb27d0 in setup() at src/main.cpp:313
  #1  0x401051fe:0x3ffb2820 in loopTask(void*) at C:/Users/test/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp:42

ELF file SHA256: d29581baf8070047

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4
[   212][E][esp32-hal-cpu.c:110] addApbChangeCallback(): duplicate func=0x40104738 arg=0x3ffbde68
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400e9ed3  PS      : 0x00060830  A0      : 0x80105201  A1      : 0x3ffb27d0
A2      : 0x3ffc4800  A3      : 0x3ffc585c  A4      : 0x3ffc4e00  A5      : 0x00000055  
A6      : 0x3ffb8874  A7      : 0x80000001  A8      : 0x800e9ed1  A9      : 0x3ffb27b0
A10     : 0x00000000  A11     : 0x00000000  A12     : 0x00000033  A13     : 0x0000005d  
A14     : 0x00061a80  A15     : 0x3ffc5024  SAR     : 0x0000001c  EXCCAUSE: 0x0000001d
EXCVADDR: 0x00000015  LBEG    : 0x40089eb4  LEND    : 0x40089eca  LCOUNT  : 0xffffffff  

Backtrace: 0x400e9ed0:0x3ffb27d0 0x401051fe:0x3ffb2820

  #0  0x400e9ed0:0x3ffb27d0 in setup() at src/main.cpp:313
  #1  0x401051fe:0x3ffb2820 in loopTask(void*) at C:/Users/test/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp:42

ELF file SHA256: d29581baf8070047

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4
[   212][E][esp32-hal-cpu.c:110] addApbChangeCallback(): duplicate func=0x40104738 arg=0x3ffbde68
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400e9ed3  PS      : 0x00060830  A0      : 0x80105201  A1      : 0x3ffb27d0
A2      : 0x3ffc4800  A3      : 0x3ffc585c  A4      : 0x3ffc4e00  A5      : 0x00000055  
A6      : 0x3ffb8874  A7      : 0x80000001  A8      : 0x800e9ed1  A9      : 0x3ffb27b0
A10     : 0x00000000  A11     : 0x00000000  A12     : 0x00000033  A13     : 0x0000005d  
A14     : 0x00061a80  A15     : 0x3ffc5024  SAR     : 0x0000001c  EXCCAUSE: 0x0000001d
EXCVADDR: 0x00000015  LBEG    : 0x40089eb4  LEND    : 0x40089eca  LCOUNT  : 0xffffffff  

Backtrace: 0x400e9ed0:0x3ffb27d0 0x401051fe:0x3ffb2820

  #0  0x400e9ed0:0x3ffb27d0 in setup() at src/main.cpp:313
  #1  0x401051fe:0x3ffb2820 in loopTask(void*) at C:/Users/test/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp:42

ELF file SHA256: d29581baf8070047

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4
[   213][E][esp32-hal-cpu.c:110] addApbChangeCallback(): duplicate func=0x40104738 arg=0x3ffbde68
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400e9ed3  PS      : 0x00060830  A0      : 0x80105201  A1      : 0x3ffb27d0  
A2      : 0x3ffc4800  A3      : 0x3ffc585c  A4      : 0x3ffc4e00  A5      : 0x00000055
A6      : 0x3ffb8874  A7      : 0x80000001  A8      : 0x800e9ed1  A9      : 0x3ffb27b0  
A10     : 0x00000000  A11     : 0x00000000  A12     : 0x00000033  A13     : 0x0000005d
A14     : 0x00061a80  A15     : 0x3ffc5024  SAR     : 0x0000001c  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000015  LBEG    : 0x40089eb4  LEND    : 0x40089eca  LCOUNT  : 0xffffffff

Backtrace: 0x400e9ed0:0x3ffb27d0 0x401051fe:0x3ffb2820

  #0  0x400e9ed0:0x3ffb27d0 in setup() at src/main.cpp:313
  #1  0x401051fe:0x3ffb2820 in loopTask(void*) at C:/Users/test/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp:42

I have checked my wiring and have checked the Firebeetle and display with other example code. Both apper to be working fine. And I made sure to load the filesystem prior to the code. Nothing seems to work.

Do you have any suggestions?

**Comment by alex-code** [comment]
The line it's referencing starts the internal file system so there might be some corruption.

If you could go to the online serial flasher we can try and get some info. https://espressif.github.io/esptool-js/

If you click connect and select the firebeetle then reply with the `VendorID`, `ProductID` and the chip detected.

One thing to try after that is the erase flash option then reflashing after. Remember to set the flash address to `0x0`

**Reply by FECRwy** [reply]
Thanks for the prompt reply.

Here is what the online serial flasher reports when connected:
esptool.js
Serial port WebSerial VendorID 0x1a86 ProductID 0x7522
Connecting.........._______..
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision 3)
Features: Wi-Fi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 94:b5:55:6b:53:44
Uploading stub...
Running stub...
Stub running...
Changing baudrate to 921600
Changed

I had tried the "Erase Flash" option before without any apparent change.

**Reply by alex-code** [reply]
Those are the ID's to expect.

Just to help narrow down the issue.
Erasing the flash and uploading `firmware.bin` downloaded from the releases page to flash address `0x0` still yields the same crash boot loop?

For curiosity I flashed the firmware to the default `0x1000` address and it crashes but it's not the same.

If you're ok with using VSCode and PlatformIO we could try an older codebase.
https://github.com/DCC-EX/EX-T3-WiFi/tree/c84f1bf9c76a2fc44f655537f18be5e9166b92cc
This is before the latest updates. If you click the green `Code` button then `Download ZIP`.
Open this in VSCode then follow the instructions for uploading.

**Reply by FECRwy** [reply]
I tried it again and got the same results. I first connected, then erased the Firebeetle and then loaded the program from a fresh download of the Firmware.bin file. Same. Here's the output:
esptool.js
Serial port WebSerial VendorID 0x1a86 ProductID 0x7522
Connecting.....
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision 3)
Features: Wi-Fi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 94:b5:55:6b:53:44
Uploading stub...
Running stub...
Stub running...
Changing baudrate to 921600
Changed
Erasing flash (this may take a while)...
Chip erase completed successfully in 11.753s
Compressed 4128768 bytes to 1103224...
Writing at 0x0... (1%)
Writing at 0x12181... (2%)
Writing at 0x1d071... (4%)
Writing at 0x2789a... (5%)
Writing at 0x2fc45... (7%)
Writing at 0x3d8b4... (8%)
Writing at 0x51335... (10%)
Writing at 0x57194... (11%)
Writing at 0x5d627... (13%)
Writing at 0x63ba7... (14%)
Writing at 0x6a612... (16%)
Writing at 0x6fc36... (17%)
Writing at 0x7555d... (19%)
Writing at 0x7b728... (20%)
Writing at 0x815b9... (22%)
Writing at 0x8732c... (23%)
Writing at 0x8d1d2... (25%)
Writing at 0x92e05... (26%)
Writing at 0x9891b... (27%)
Writing at 0x9dd1e... (29%)
Writing at 0xa310d... (30%)
Writing at 0xa838a... (32%)
Writing at 0xad7c4... (33%)
Writing at 0xb2a71... (35%)
Writing at 0xb7a96... (36%)
Writing at 0xbcf87... (38%)
Writing at 0xc21c3... (39%)
Writing at 0xc7540... (41%)
Writing at 0xcd0b8... (42%)
Writing at 0xd28d1... (44%)
Writing at 0xd7c60... (45%)
Writing at 0xdcee5... (47%)
Writing at 0xe22ff... (48%)
Writing at 0xe7936... (50%)
Writing at 0xecfe1... (51%)
Writing at 0xf2767... (52%)
Writing at 0xf870c... (54%)
Writing at 0xfe138... (55%)
Writing at 0x104f37... (57%)
Writing at 0x10df44... (58%)
Writing at 0x116ce2... (60%)
Writing at 0x11c5fc... (61%)
Writing at 0x124039... (63%)
Writing at 0x12b76f... (64%)
Writing at 0x13430c... (66%)
Writing at 0x13c238... (67%)
Writing at 0x1414a4... (69%)
Writing at 0x149617... (70%)
Writing at 0x14ee46... (72%)
Writing at 0x15400d... (73%)
Writing at 0x1595c3... (75%)
Writing at 0x2152bd... (76%)
Writing at 0x231388... (77%)
Writing at 0x245d0b... (79%)
Writing at 0x2609af... (80%)
Writing at 0x268b00... (82%)
Writing at 0x270e90... (83%)
Writing at 0x279559... (85%)
Writing at 0x2816d2... (86%)
Writing at 0x289d25... (88%)
Writing at 0x2922c5... (89%)
Writing at 0x29a72d... (91%)
Writing at 0x2a2875... (92%)
Writing at 0x2aaaf2... (94%)
Writing at 0x2b2f7e... (95%)
Writing at 0x2bb3ff... (97%)
Writing at 0x2c3887... (98%)
Writing at 0x2cbd3e... (100%)
Wrote 4128768 bytes (1103224 compressed) at 0x0 in 30.562 seconds.
Hash of data verified.
Leaving...

I'll give the VSCode and PlatformIO another try.

**Reply by alex-code** [reply]
The upload output above is the same as I get.

In VSCode could you try commenting (add two forward slashes) line `main.cpp#313` then uploading to see if it gets further, it'll still error but will be different.

I don't know if any hardware issues could cause this.
If it's not too assembled you could strip it back down and just try the firebeetle on its own. It will error without the display but will be different.

One last thing to try would be creating a basic project that has a single file in the flash file system that it tries to read.

**Comment by FECRwy** [comment]
Same result.

**Comment by FECRwy** [comment]
I tried what you suggested and still no luck. Do you think I should order some new hardware?

**Reply by alex-code** [reply]
Were you able to try a new basic project that has a simple SPIFFS file system and tries to mount it?
Been googling and can't find any similar issues.

**Reply by FECRwy** [reply]
Can you suggest an example? All i've done is use the Arduino IDE to load an example sketch called UI_keyPad for the Firebeetle using their library DFRobot_GDL -> TouchIL9488_320x480 -> UI_keyPad

**Reply by alex-code** [reply]
There's a test `ino` file if you want to use Arduino IDE or if you want to use PlatformIO in VSCode you can create a new project and copy the code.

https://github.com/espressif/arduino-esp32/blob/master/libraries/SPIFFS/examples/SPIFFS_Test/SPIFFS_Test.ino

This will try to load the flash file system and do some file operations.

**Reply by FECRwy** [reply]
Uploaded the example in the Arduino IDE. Load was successful but the only thing that happened was the display went all white. Nothing else was noted. May be time to replace the Firebeetle?

**Reply by alex-code** [reply]
It'll output a load of text to the serial monitor, if you open that does it show messages about creating and reading files.

Were you able to strip it down so it's just the firebeetle board on its own with the throttle code?
Wondering if there's a short.

**Reply by FECRwy** [reply]
Sorry, I hadn't looked at the monitor. Here's a dump of the output to the Serial Monitor:
15:23:35.318 -> rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
15:23:35.318 -> configsip: 0, SPIWP:0xee
15:23:35.318 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
15:23:35.373 -> mode:DIO, clock div:1
15:23:35.373 -> load:0x3fff0030,len:4832
15:23:35.373 -> load:0x40078000,len:16460
15:23:35.373 -> load:0x40080400,len:4
15:23:35.373 -> load:0x40080404,len:3504
15:23:35.373 -> entry 0x400805cc
15:23:35.596 -> Listing directory: /
15:23:35.736 -> Writing file: /hello.txt
15:23:36.015 -> - file written
15:23:36.144 -> Appending to file: /hello.txt
15:23:36.144 -> - message appended
15:23:36.280 -> Reading file: /hello.txt
15:23:36.280 -> - read from file:
15:23:36.280 -> Hello World!
15:23:36.280 -> Renaming file /hello.txt to /foo.txt
15:23:36.326 -> - file renamed
15:23:36.326 -> Reading file: /foo.txt
15:23:36.326 -> - read from file:
15:23:36.326 -> Hello World!
15:23:36.326 -> Deleting file: /foo.txt
15:23:36.326 -> - file deleted
15:23:36.326 -> Testing file I/O with /test.txt
15:23:36.484 -> - writing................................................................
15:24:40.240 ->  - 1048576 bytes written in 63756 ms
15:24:40.315 -> - reading................................
15:24:40.642 -> - 524288 bytes read in 359 ms
15:24:40.642 -> Deleting file: /test.txt
15:24:42.185 -> - file deleted
15:24:42.185 -> Test complete

This is with just the Firebeetle and the display. Even with this, I got just a flashing screen when I tried the DCCEX T3 code. I think I will order another Firebeetle and see if that works. I reall y don't see where a short would be from.

**Reply by alex-code** [reply]
Really strange.
Just to double check have you got a FireBeetle 2 ESP32-E board as there's one with a different chip but it's also labelled as a FireBeetle 2.

**Reply by FECRwy** [reply]
FireBeetle ESP32-E V1.0 on the back.

**Reply by alex-code** [reply]
That is the one, not sure why the SPIFFS test works but the throttle code doesn't.

**Reply by FECRwy** [reply]
I can't figure it out either. I checked all of my connections and all seem fine. When I ran other code, it appeared to function okay as well. This one got me stumped so that is why I asked for help. I've decided to get another FireBeetle in order to see if that is the problem. Perhaps something internal. Anyway, I'll let you know once I've received the new one and test it. Thanks for your help.
